src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -1112,6 +1112,109 @@ protected static ImmutableArray<string> CheckTupleElementNames(int cardinality,
+        /// to be an <see cref="ITypeSymbol"/> or <see cref="IAssemblySymbol"/>.
+        /// </summary>
+        public bool IsSymbolAccessibleWithin(
+            ISymbol symbol,
+            ISymbol within,
+            ITypeSymbol throughType = null)
+        {
+            if (symbol is null)
+            {
+                throw new ArgumentNullException(nameof(symbol));
+            }
+
+            if (within is null)
+            {
+                throw new ArgumentNullException(nameof(within));
+            }
+
+            if (!(within is INamedTypeSymbol || within is IAssemblySymbol))
+            {
+                throw new ArgumentException(CodeAnalysisResources.IsSymbolAccessibleBadWithin, nameof(within));
+            }
+
+            checkInCompilation(symbol, nameof(symbol));
+            checkInCompilation(within, nameof(within));
+            if (!(throughType is null))
+            {
+                checkInCompilation(throughType, nameof(throughType));
+            }
+
+            return IsSymbolAccessibleWithinCore(symbol, within, throughType);
+
+            void checkInCompilation(ISymbol s, string parameterName)
+            {
+                var containingAssembly = computeContainingAssembly(s);
+                if (!assemblyIsOk(containingAssembly))
+                {
+                    throw new ArgumentException(string.Format(CodeAnalysisResources.IsSymbolAccessibleWrongAssembly, parameterName), parameterName);
+                }
+            }
+
+            bool assemblyIsOk(IAssemblySymbol a)
+            {
+                if (a == this.Assembly)
+                {
+                    return true;
+                }
+
+                foreach (var reference in this.References)
+                {
+                    var moduleOrAssembly = this.CommonGetAssemblyOrModuleSymbol(reference);
+                    var assembly = moduleOrAssembly as IAssemblySymbol ?? moduleOrAssembly.ContainingAssembly;
+                    if (a == assembly)
+                    {
+                        return true;
+                    }
+                }
+
+                return false;
+            }
+
+            IAssemblySymbol computeContainingAssembly(ISymbol s)
+            {
+                while (true)
+                {
+                    switch (s.Kind)
+                    {
+                        case SymbolKind.Assembly:
+                            return (IAssemblySymbol)s;
+                        case SymbolKind.PointerType:
+                            s = ((IPointerTypeSymbol)s).PointedAtType;
+                            continue;
+                        case SymbolKind.ArrayType:
+                            s = ((IArrayTypeSymbol)s).ElementType;
+                            continue;
+                        case SymbolKind.Alias:
+                            s = ((IAliasSymbol)s).Target;
```

Comment:
>((IAliasSymbol)s).Target [](start = 32, length = 24)

Is this never null?

Reply:
> Is this never null?

The compilers do no expose alias symbols with null targets.

---
In reply to: [192574937](https://github.com/dotnet/roslyn/pull/27312#discussion_r192574937) [](ancestors = 192574937)